src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt

```diff
@@ -40,26 +40,23 @@ class PDU private constructor() : PlatformDependentUtilsBase() {
         val projectPath = psiFile.project.basePath
             ?: throw IllegalStateException("No Project base Path found for file $psiFile")
         // `ignoreCase = true` needed due to https://youtrack.jetbrains.com/issue/IDEA-268081
-        val projectBasedPath = absolutePath.replaceFirst(projectPath, "", ignoreCase = true)
-        if (projectBasedPath == absolutePath) {
-            throw IllegalStateException("projectBasedPath is not valid:\n" +
-                "  psiFile.virtualFile.path = $absolutePath\n" +
-                "  psiFile.project.basePath = $projectPath")
-        }
-        return projectBasedPath
+        return absolutePath.replaceFirst(projectPath, "", ignoreCase = true)
```

Comment:
Ignoring the case could lead to bugs on MacOS and Linux, as a file with different case would be a different file. How about covering this scenario with a test, so we are certain it works?

Reply:
ignoring the case was added exactly to avoid bugs on Mac/Linux :) See comment above the line:
`ignoreCase = true` needed due to https://youtrack.jetbrains.com/issue/IDEA-268081

Reply:
I still don't understand it - I looked at the issue, and github is reported with different case, it seems. Still, different case means, different file/directory on Linux/macOs, so it should not be treated case-insensitive. For me the question is why in the you track issue the virtual file is reporting a different path, and if that cannot be helped, if we should differentiate between platforms.

Reply:
The problem is that on Mac/Linux we have different case for the **same** dir returned from those api:
PsiFile.project.basePath
PsiFile.virtualFile.path
And we use the text of `basePath` to subtract it from full-file-path to get project-related-path.
If we don't do it case insensitive then basePath is treated as different paths and will not be properly subtracted.
You're right, in some edge cases: two files with the same basePath (just different in case) that might cause an issue. But it seems to be not realist in real world so far.
PS we are going to get off project.basePath anyway (due to issue with multi-module projects that have no basePath) so would leave it as is for now, and it will be fixed automatically after project.basePath issue fixed.

Reply:
Okay, let's go ahead. If we chuck project.basePath anyway, no use to cater to that special edge case right now. It's weird though, that project.basePath is not returning the case-correct path.